libs/designsystem/src/lib/components/segmented-control/segmented-control.component.spec.ts

```diff
@@ -27,92 +27,129 @@ describe('SegmentedControlComponent', () => {
       checked: true,
       id: 'second',
     },
+    {
+      text: 'Third item',
+      checked: false,
+      id: 'third',
+    },
   ];
 
-  beforeEach(async(() => {
-    TestBed.configureTestingModule({
-      declarations: [
-        SegmentedControlComponent,
-        IonSegment,
-        MockComponents(ChipComponent, BadgeComponent, IonSegmentButton),
-        MockDirective(ThemeColorDirective),
-      ],
-    }).compileComponents();
-  }));
+  let spectator: SpectatorHost<SegmentedControlComponent>;
+
+  const createHost = createHostFactory({
+    component: SegmentedControlComponent,
+    declarations: [
+      MockComponents(ChipComponent, BadgeComponent),
+      MockDirective(ThemeColorDirective),
+    ],
+    imports: [IonicModule.forRoot({ mode: 'ios' })],
+  });
 
   beforeEach(() => {
-    fixture = TestBed.createComponent(SegmentedControlComponent);
-    component = fixture.componentInstance;
-    component.items = items;
-    fixture.detectChanges();
+    spectator = createHost(
+      `<kirby-segmented-control [items]="items">
+       </kirby-segmented-control>`,
+      {
+        hostProps: {
+          items: items,
+        },
+      }
+    );
+    component = spectator.component;
+    onSegmentSelectSpy = spyOn(component.segmentSelect, 'emit');
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
+  it('should not emit segmentSelect when created', () => {
+    expect(onSegmentSelectSpy).not.toHaveBeenCalled();
```

Comment:
This may report a "false positive", since you're spying on the `component.segmentSelect.emit`-method, after it's been created (and `ngOnInit` has been run).

Reply:
Good point - removed 🔥